### DISCUSSION TITLE
Will running prisma.$disconnect interrupt existing queries?

### DISCUSSION THREAD

**Discussion by vedantroy** [prisma/prisma#19549] [Q&A] [answered]
If I'm running `prisma.$disconnect`, will it wait for existing queries to finish?

**Comment by ludralph** [comment, marked as answer]
Hi @vedantroy 👋 

When you call prisma.$disconnect() in a Prisma client, it will immediately close the underlying database connection and prevent any new queries or transactions from being executed. However, it will not wait for existing queries to finish before disconnecting. If there are any ongoing queries or transactions at the time of disconnection, they will be interrupted, and their results may be lost. It's important to ensure that you've handled any necessary error handling or rollback logic to handle potential interruptions gracefully. If you need to wait for existing queries to finish before disconnecting, you'll need to implement your own mechanism to track the completion of those queries and ensure they finish before calling $disconnect(). See the [disconnect guide](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#disconnect)
